component/core.py

```diff
@@ -346,14 +346,58 @@ def component_by_name(self, name, model_name=None):
             work_context = self.work_on(model_name)
         return component_class(work_context)
 
-    def _lookup_components(self, usage=None, model_name=None):
+    def _lookup_components(self, usage=None, model_name=None, **kw):
         component_classes = self.components_registry.lookup(
             self.collection._name, usage=usage, model_name=model_name
         )
+        matching_components = []
+        for cls in component_classes:
+            try:
+                matching = cls._component_match(
+                    self, usage=usage, model_name=model_name, **kw
+                )
+            except TypeError as err:
+                # Backward compat
```

Comment:
I think we can drop this

Reply:
I would agree with you if this was the migration of the module, but component was already on 14, could be implementations with the old behavior, isn't it?

Reply:
oh yeah... well, it's unlikely that someone apart from @guewen customized it  but... ok, let's stay safe and do it on v15 :+1: